api/v1beta1/azuremachine_types.go

```diff
@@ -142,6 +142,11 @@ type AzureMachineSpec struct {
 	// The primary interface will be the first networkInterface specified (index 0) in the list.
 	// +optional
 	NetworkInterfaces []NetworkInterface `json:"networkInterfaces,omitempty"`
+
+	// CapacityReservationGroupID specifies the capacity reservation group resource id that should be
+	// used for allocating the virtual machine.
+	// +optional
+	CapacityReservationGroupID *string `json:"capacityReservationGroupID,omitempty"`
```

Comment:
Assuming this is in the format of an Azure resource ID, I think it might be good to add some webhook validation for it similar to how it is done for the Provider ID: https://github.com/willie-yao/cluster-api-provider-azure/blob/main/api/v1beta1/azuremachine_validation.go#L138

Reply:
I have added the webhook validation to check if `CapacityReservationGroupID` is a valid Azure resource ID.

Reply:
It would be good to include in the godoc an explanation of what the requirements are for the ID, which characters are allowed, length wise as well, what's the minimum and maximum?

Would also be good to explain the immutability as part of the godoc